### PR TITLE
feat: Simple icing state simulation 

### PR DESCRIPTION
**Pull request by crocket63**
Fixes #[issue_no]

## Summary of Changes

Adds a small implementation so we can make any part taking ice, and be deiced naturally or actively.

Used on icing stick in front of 380 windshield.

## Additional context

Discord username (if different from GitHub):

## Testing instructions

No A320 impact

## How to download the PR for QA

Every new commit to this PR will cause a new A32NX artifact to be created, built, and uploaded.

1. Make sure you are signed in to GitHub
1. Click on the **Checks** tab on the PR
1. On the left side, click on the bottom **PR** tab
1. Click on the **A32NX** download link at the bottom of the page

### CODE REVIEW

File: fbw-common/src/wasm/systems/systems/src/icing_state/mod.rs [hunk 1]
```diff
@@ -0,0 +1,387 @@
...
+        } else {
+            0.
+        };
+        Ratio::new::<ratio>(raw_icing_ratio.clamp(0., 1.))
+    }
+
+    fn natural_deicing_rate(context: &UpdateContext) -> Ratio {
+        if context.ambient_temperature().get::<degree_celsius>() > 0. {
+            Ratio::new::<ratio>(
+                (0.25 * context.ambient_temperature().get::<degree_celsius>().sqrt()).clamp(0., 2.),
```
[the remark this answers is not in the document]
Comment (crocket63) [reply fbw-common/src/wasm/systems/systems/src/icing_state/mod.rs:126]: Do I look like I remember? :D  So idea is passive deicing is for "standard" temperature. So having 2 ratio means if you end up with a crazy 50°C temperature in Dubai, you'll get 2 times faster deicing.  Absolutely not physical based obviously but that's a start :p

File: fbw-a32nx/src/wasm/systems/a320_systems_wasm/src/lib.rs [hunk 2]
```diff
@@ -196,6 +196,8 @@ async fn systems(mut gauge: msfs::Gauge) -> Result<(), Box<dyn Error>> {
     .provides_aircraft_variable("AIRSPEED MACH", "Mach", 0)?
     .provides_aircraft_variable("AIRSPEED TRUE", "Knots", 0)?
     .provides_aircraft_variable("AMBIENT DENSITY", "Slugs per cubic feet", 0)?
+    .provides_aircraft_variable("AMBIENT IN CLOUD", "Bool", 0)?
+    .provides_aircraft_variable("AMBIENT PRECIP RATE", "millimeters of water", 0)?
```
Comment (BlueberryKing) [inline fbw-a32nx/src/wasm/systems/a320_systems_wasm/src/lib.rs:200]: Do these not need to be part of `a380_systems_wasm/src/lib.rs` as well?

Comment (crocket63) [reply fbw-a32nx/src/wasm/systems/a320_systems_wasm/src/lib.rs:200]: Pretty sure they are !?

Comment (BlueberryKing) [reply fbw-a32nx/src/wasm/systems/a320_systems_wasm/src/lib.rs:200]: Ah, don't ask me what I was looking at, found them now, sorry!

**Review by BlueberryKing — APPROVED** [review]

**Pull request merged**